pkgs/servers/http/bozohttpd/default.nix

```
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchurl
+, bmake
+, groff
+, inetutils
+, wget
+, openssl
+, userSupport ? true
+, cgiSupport ? true
+, dirIndexSupport ? true
+, dynamicContentSupport ? true
+, sslSupport ? true
+, luaSupport ? true
+, lua
+, htpasswdSupport ? true
```

Do we need all those options even? Can we just enable them by default?

```suggestion
, lua
```

---

We could, and I'm enabling them all by default, but I figured if the upstream developer exposes and supports these knobs, it's easy enough for us to do so as well.

---

We usually do not do this and just enable all common defaults people usually want to use.

---

I feel like if these flags are there upstream, there's probably someone who uses them, especially as most of them are "opt-outs" rather than "opt-ins" - eg. lua support is enabled by default, but you can explicitly disable it. Especially in a web server, I can see why you'd would want to disable functionality that you're not using, and reduce its attack surface.

Is there a reason why you'd like to see them removed?

---

> Is there a reason why you'd like to see them removed?

They add extra complexity and most of the time they are not tested while updating which causes them to bitrot and break over time especially if you use a complex combination of them.

---

We test all of these upstream because we have a "small" target that builds an extremely small bozohttpd for extremely low memory machines. The only one that affects anything other than which fragmwnts of code are included is the "lua" and "ssl" options, which change dependencies. Since bozohttpd has to build entirely from the NetBSD base system, it is extremely unlikely that any new requirements would be added. I don't see how supporting them introduces any complexity.

---

What about a `bozohttpdMinimal` top level attribute with all optional features disabled by default? This way the “testing this still builds” concern should be alleviated a bit.

---

Ensuring that packages do not bitrot when updated is firmly the maintainer's responsibility. If you're concerned for my workload, it's appreciated, but not necessary - I wouldn't have added these options if I wasn't confident that they would be maintainable. As nia says, they are unlikely to change.

But they could indeed use a CI pass, so I've added another `minimal` option and a `bozohttpd-minimal` top-level package, to roughly mirror the netbsd `small` target.

---

@SuperSandro2000 Should I take your silence as that this has adequately addressed your concerns?